File: FrontendApp/app/services/trainingType.service.js
function trainingTypeService($http) {
    var service = this;

    service.createTrainingType = function (trainingType) {
        return $http({
            method : "POST",
            url : "http://localhost:8081/rest/training-types",
            data : trainingType
        }).then(function (response) {
            return response.data;
        });
    };

    service.updateTrainingType = function (trainingType) {
        return $http({
            method : "PUT",
            url : "http://localhost:8081/rest/training-types/" + trainingType.id,
            data : trainingType
        }).then(function (response) {
            return response.data;
        });
    };

    service.deleteTrainingType = function (trainingTypeId) {
        return $http({
            method : "DELETE",
            url : "http://localhost:8081/rest/training-types/" + trainingTypeId
        });
    };
    
    service.getTrainingTypeById = function (trainingTypeId) {
        return $http({
            method : "GET",
            url : "http://localhost:8081/rest/training-types/" + trainingTypeId
        }).then(function (response) {
            return response.data;
        });
    };
}

File: BackendServer/src/main/java/by/bsuir/nesenchuk/service/TrainingTypeService.java
package by.bsuir.nesenchuk.service;

import by.bsuir.nesenchuk.entity.TrainingType;

import java.util.List;

public interface TrainingTypeService {
    TrainingType createTrainingType(TrainingType trainingType);
    TrainingType updateTrainingType(TrainingType trainingType);
    void deleteTrainingType(Long id);
    List<TrainingType> getTrainingTypes();
}
